### PR TITLE
[luci-interpreter] GRU kernel

### PR DESCRIPTION
**Pull request by chunseoklee**
- Enable CircleGRU operation in luci-interpreter
- Overall implemenatation is borrowed from onert-micro 2.0

draft : https://github.com/Samsung/ONE/pull/14237
issue : https://github.com/Samsung/ONE/issues/12263

ONE-DCO-1.0-Signed-off-by: Chunseok Lee <chunseok.lee@samsung.com>

### CODE REVIEW

File: compiler/luci-interpreter/pal/linux/KernelsToBuild.lst [hunk 1]
```diff
@@ -26,6 +26,7 @@ REGISTER_KERNEL(FullyConnected)
 REGISTER_KERNEL(Gather)
 REGISTER_KERNEL(Gelu)
 REGISTER_KERNEL(Greater)
+REGISTER_KERNEL(GRU)
 REGISTER_KERNEL(GreaterEqual)
```
Comment (ys44kim) [inline compiler/luci-interpreter/pal/linux/KernelsToBuild.lst:30]: (Optional)
I think It looks good to apply alphabetically.
```suggestion
REGISTER_KERNEL(Greater)
REGISTER_KERNEL(GreaterEqual)
REGISTER_KERNEL(GRU)
```

Comment (chunseoklee) [reply compiler/luci-interpreter/pal/linux/KernelsToBuild.lst:29]: updated

File: compiler/luci-interpreter/src/kernels/GRU.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+void GRU::configure()
+{
+  auto hidden_hidden_shape = getTensorShape(hidden_hidden());
+  auto hidden_input_shape = getTensorShape(hidden_input());
+  LUCI_INTERPRETER_CHECK(hidden_hidden_shape.Dims(0) == hidden_input_shape.Dims(0));
+
+  auto output_shape = getTensorShape(output());
+  auto state_shape = getTensorShape(state());
```
Comment (ys44kim) [inline compiler/luci-interpreter/src/kernels/GRU.cpp:44]: May I ask what `output_shape`, `state_shape`  were added for?

File: compiler/luci-interpreter/pal/linux/PALGRU.h [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    calculateGRU(input_data, weight_input_data, weight_hidden_data, bias_input_data,
+                 bias_hidden_data, output_data, input_shape, output_shape, weight_input_shape,
+                 weight_hidden_shape, output_input_data, output_hidden_data, output_shape_fc);
+    input_data += input_shape.Dims(2);
+  }
+}
+
+} // namespace luci_interpreter_pal
+
+#endif // ONERT_MICRO_EXECUTE_PAL_GRU_COMMON_H
```
Comment (seanshpark) [inline compiler/luci-interpreter/pal/linux/PALGRU.h:173]:
```suggestion
#endif // LUCI_INTERPRETER_PAL_GRU_H
```

File: compiler/luci-interpreter/src/kernels/GRU.cpp (hunk 2 above)
Comment (chunseoklee) [reply compiler/luci-interpreter/src/kernels/GRU.cpp:44]: I will updated soon ( it forgot to remove this in the draft)

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by ys44kim — APPROVED** [review]
LGTM

**Pull request merged**